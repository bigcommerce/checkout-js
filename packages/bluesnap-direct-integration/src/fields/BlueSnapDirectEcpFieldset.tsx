import { LanguageService } from '@bigcommerce/checkout-sdk';
import React, { FunctionComponent, memo } from 'react';

import BlueSnapDirectNumberField from './BlueSnapDirectNumberField';
import BlueSnapDirectSelectField from './BlueSnapDirectSelectField';

export interface BlueSnapDirectEcpFieldsetProps {
    language: LanguageService;
    useFloatingLabel?: boolean;
}

const BlueSnapDirectEcpFieldset: FunctionComponent<BlueSnapDirectEcpFieldsetProps> = ({
    language,
    useFloatingLabel,
}) => {
    const options = {
        helperLabel: language.translate('payment.bluesnap_direct_account_type_select.label'),
        items: [
            {
                label: language.translate(
                    'payment.bluesnap_direct_account_type_select.option_consumer_checking',
                ),
                value: 'CONSUMER_CHECKING',
            },
            {
                label: language.translate(
                    'payment.bluesnap_direct_account_type_select.option_consumer_savings',
                ),
                value: 'CONSUMER_SAVINGS',
            },
            {
                label: language.translate(
                    'payment.bluesnap_direct_account_type_select.option_corporate_checking',
                ),
                value: 'CORPORATE_CHECKING',
            },
            {
                label: language.translate(
                    'payment.bluesnap_direct_account_type_select.option_corporate_savings',
                ),
                value: 'CORPORATE_SAVINGS',
            },
        ],
    };

    return (
        <>
            <BlueSnapDirectNumberField
                labelContent={language.translate('payment.bluesnap_direct_account_number.label')}
                maxLength={17}
                name="accountNumber"
                useFloatingLabel={useFloatingLabel}
            />

            <BlueSnapDirectNumberField
                labelContent={language.translate('payment.bluesnap_direct_routing_number.label')}
                maxLength={9}
                name="routingNumber"
                useFloatingLabel={useFloatingLabel}
            />

            <BlueSnapDirectSelectField
                labelContent={language.translate('payment.bluesnap_direct_account_type.label')}
                name="accountType"
                options={options}
                useFloatingLabel={useFloatingLabel}
            />
        </>
    );
};

export default memo(BlueSnapDirectEcpFieldset);
