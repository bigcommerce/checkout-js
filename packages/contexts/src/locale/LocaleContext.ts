import type { CurrencyService, LanguageService } from '@bigcommerce/checkout-sdk';
import { createContext } from 'react';

export interface LocaleContextType {
    language: LanguageService;
    date?: {
        inputFormat: string;
    };
    currency?: CurrencyService;
}

export const LocaleContext = createContext<LocaleContextType | undefined>(undefined);
