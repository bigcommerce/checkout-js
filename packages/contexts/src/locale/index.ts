export { LocaleContext, type LocaleContextType } from './LocaleContext';
export { default as LocaleProvider } from './LocaleProvider';
export { useLocale } from './useLocale';
