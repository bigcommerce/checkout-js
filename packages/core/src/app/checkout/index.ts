export { RenderCheckout, RenderCheckoutOptions } from './renderCheckout';
export { default as CheckoutContext, CheckoutContextProps } from './CheckoutContext';
export {
    default as CheckoutProvider,
    CheckoutProviderProps,
    CheckoutProviderState,
} from './CheckoutProvider';
export { default as withCheckout, WithCheckoutProps } from './withCheckout';
export { default as CheckoutSupport } from './CheckoutSupport';
export { default as NoopCheckoutSupport } from './NoopCheckoutSupport';
export { default as navigateToOrderConfirmation } from './navigateToOrderConfirmation';
