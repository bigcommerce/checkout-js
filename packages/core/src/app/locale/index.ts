export { default as LocaleContext, LocaleContextType } from './LocaleContext';
export { default as createLocaleContext } from './createLocaleContext';
export { default as getDefaultTranslations } from './getDefaultTranslations';
export { default as getLanguageService, InitializeLanguageService } from './getLanguageService';
export { default as masterpassFormatLocale } from './masterpassFormatLocale';
export { default as withCurrency, WithCurrencyProps } from './withCurrency';
export { default as withLanguage, WithLanguageProps } from './withLanguage';
export { default as withDate, WithDateProps } from './withDate';
export { default as LanguageWindow, isLanguageWindow } from './LanguageWindow';
export { default as LocaleProvider } from './LocaleProvider';
export { default as TranslatedHtml, TranslatedHtmlProps } from './TranslatedHtml';
export { default as TranslatedLink, TranslatedLinkProps } from './TranslatedLink';
export { default as TranslatedString, TranslatedStringProps } from './TranslatedString';
