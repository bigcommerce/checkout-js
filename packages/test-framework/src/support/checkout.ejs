<!DOCTYPE html>
<html>
    <head>
        <title>Local Testing Environment</title>
        <base href="/" target="_top"/>
        <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1" />
        <link href="https://fonts.googleapis.com/css?family=Karla:400|Montserrat:400,500,700&display=swap" rel="stylesheet" />
    </head>
    <body>
    <header class="checkoutHeader optimizedCheckout-header">
        <div class="checkoutHeader-content">
            <h1 class="is-srOnly">Checkout</h1>
            <h2 class="checkoutHeader-heading">
                <a class="checkoutHeader-link" href="<%= storeUrl %>/">
                    <span class="header-logo-text">Local Testing Environment</span>
                </a>
            </h2>
        </div>
    </header>
    <div class="ng-checkout-container" id="micro-app-ng-checkout">
        <div class="ng-checkout-view" data-ui-view></div>
    </div>
    <script type="text/javascript">
        window.checkoutConfig = {
            checkoutId: '<%= checkoutId %>',
            containerId: 'micro-app-ng-checkout',
        };
    </script>
    <script src="<%= localhostUrl %>/auto-loader.js" type="text/javascript"></script>
    <script type="text/javascript">
        (function() {
            window.checkout.renderCheckout({
                checkoutId: '<%= checkoutId %>',
                containerId: 'micro-app-ng-checkout',
                sentryConfig: null,
            })
        })();
    </script>
    </body>
</html>
