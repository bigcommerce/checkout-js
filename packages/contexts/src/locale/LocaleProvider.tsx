import {
    type CheckoutService,
    createCurrencyService,
    type LanguageService,
    type StoreConfig,
} from '@bigcommerce/checkout-sdk/essential';
import { memoizeOne } from '@bigcommerce/memoize';
import React, { type ReactNode, useEffect, useMemo, useState } from 'react';

import { LocaleContext } from './LocaleContext';

export interface LocaleProviderProps {
    languageService: LanguageService;
    checkoutService: CheckoutService;
    children?: ReactNode;
}

const LocaleProvider: React.FC<LocaleProviderProps> = ({
    checkoutService,
    children,
    languageService,
}) => {
    const [config, setConfig] = useState<StoreConfig | undefined>();

    const getContextValue = memoizeOne((storeConfig?: StoreConfig) => {
        return {
            currency: storeConfig ? createCurrencyService(storeConfig) : undefined,
            date: storeConfig
                ? {
                      inputFormat: storeConfig.inputDateFormat,
                  }
                : undefined,
            language: languageService,
        };
    });

    useEffect(() => {
        const unsubscribe = checkoutService.subscribe(
            ({ data }) => {
                setConfig(data.getConfig());
            },
            ({ data }) => data.getConfig(),
        );

        return () => {
            unsubscribe();
        };
    }, [checkoutService]);

    const contextValue = useMemo(() => getContextValue(config), [config, getContextValue]);

    return <LocaleContext.Provider value={contextValue}>{children}</LocaleContext.Provider>;
};

export default LocaleProvider;
