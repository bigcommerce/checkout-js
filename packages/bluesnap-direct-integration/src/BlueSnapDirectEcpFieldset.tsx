import { LanguageService } from '@bigcommerce/checkout-sdk';
import React, { FunctionComponent, memo } from 'react';

import { CheckboxFormField, Fieldset, Legend } from '@bigcommerce/checkout/ui';

import BlueSnapDirectNumberField from './BlueSnapDirectNumberField';
import BlueSnapDirectSelectField from './BlueSnapDirectSelectField';

export interface BlueSnapDirectEcpFieldsetProps {
    language: LanguageService;
    useFloatingLabel?: boolean;
    onPermissionChange: (shopperPermission: boolean) => void;
}

const BlueSnapDirectEcpFieldset: FunctionComponent<BlueSnapDirectEcpFieldsetProps> = ({
    language,
    useFloatingLabel,
    onPermissionChange,
}) => {
    const options = {
        helperLabel: 'Select an account type',
        items: [
            { label: 'Consumer checking', value: 'CONSUMER_CHECKING' },
            { label: 'Consumer savings', value: 'CONSUMER_SAVINGS' },
            { label: 'Corporate checking', value: 'CORPORATE_CHECKING' },
            { label: 'Corporate savings', value: 'CORPORATE_SAVINGS' },
        ],
    };

    return (
        <Fieldset
            legend={<Legend hidden>Electronic Check (ACH/ECP)</Legend>}
            style={{ paddingBottom: '1rem' }}
        >
            <BlueSnapDirectNumberField
                labelContent={language.translate('payment.bluesnap_direct_account_number.label')}
                maxLength={17}
                name="accountNumber"
                useFloatingLabel={useFloatingLabel}
            />

            <BlueSnapDirectNumberField
                labelContent={language.translate('payment.bluesnap_direct_routing_number.label')}
                maxLength={9}
                name="routingNumber"
                useFloatingLabel={useFloatingLabel}
            />

            <BlueSnapDirectSelectField
                labelContent={language.translate('payment.bluesnap_direct_account_type.label')}
                name="accountType"
                options={options}
                useFloatingLabel={useFloatingLabel}
            />

            <CheckboxFormField
                labelContent={language.translate('payment.bluesnap_direct_permission')}
                name="shopperPermission"
                onChange={onPermissionChange}
            />
        </Fieldset>
    );
};

export default memo(BlueSnapDirectEcpFieldset);
