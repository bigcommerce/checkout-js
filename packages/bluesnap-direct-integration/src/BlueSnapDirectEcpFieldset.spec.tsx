import { createLanguageService } from '@bigcommerce/checkout-sdk';
import { mount, render } from 'enzyme';
import { Formik } from 'formik';
import { noop } from 'lodash';
import React from 'react';

import { createLocaleContext, LocaleContext } from '@bigcommerce/checkout/locale';
import { getStoreConfig } from '@bigcommerce/checkout/test-utils';

import BlueSnapDirectEcpFieldset, {
    BlueSnapDirectEcpFieldsetProps,
} from './BlueSnapDirectEcpFieldset';

describe('BlueSnapDirectEcpFieldset', () => {
    let initialValues: { [key: string]: unknown };
    let options: BlueSnapDirectEcpFieldsetProps;

    beforeEach(() => {
        initialValues = {
            accountNumber: '',
            routingNumber: '',
        };
        options = {
            language: createLanguageService(),
            onPermissionChange: jest.fn(),
        };
    });

    it('matches snapshot', () => {
        expect(
            render(
                <Formik initialValues={initialValues} onSubmit={noop}>
                    <LocaleContext.Provider value={createLocaleContext(getStoreConfig())}>
                        <BlueSnapDirectEcpFieldset {...options} />
                    </LocaleContext.Provider>
                </Formik>,
            ),
        ).toMatchSnapshot();
    });

    it('should call onPermissionChange callback', () => {
        const component = mount(
            <Formik initialValues={initialValues} onSubmit={noop}>
                <BlueSnapDirectEcpFieldset {...options} />
            </Formik>,
        );

        component
            .find('input[name="shopperPermission"]')
            .simulate('change', { target: { value: true, name: 'shopperPermission' } })
            .update();

        expect(options.onPermissionChange).toHaveBeenCalledWith(true);
    });
});
