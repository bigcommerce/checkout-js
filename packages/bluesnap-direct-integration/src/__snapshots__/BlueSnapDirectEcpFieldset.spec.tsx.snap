// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlueSnapDirectEcpFieldset matches snapshot 1`] = `
<fieldset
  class="form-fieldset"
  style="padding-bottom:1rem"
>
  <legend
    class="form-legend is-srOnly"
  >
    Electronic Check (ACH/ECP)
  </legend>
  <div
    class="form-body"
  >
    <div
      class="form-field"
    >
      <label
        class="form-label optimizedCheckout-form-label"
        for="accountNumber"
        id="accountNumber-label"
      >
        optimized_checkout.payment.bluesnap_direct_account_number.label
      </label>
      <input
        aria-labelledby="accountNumber-label accountNumber-field-error-message"
        autocomplete="cc-number"
        class="form-input optimizedCheckout-form-input"
        data-test="accountNumber-text"
        id="accountNumber"
        maxlength="17"
        name="accountNumber"
        type="tel"
        value=""
      />
    </div>
    <div
      class="form-field"
    >
      <label
        class="form-label optimizedCheckout-form-label"
        for="routingNumber"
        id="routingNumber-label"
      >
        optimized_checkout.payment.bluesnap_direct_routing_number.label
      </label>
      <input
        aria-labelledby="routingNumber-label routingNumber-field-error-message"
        autocomplete="cc-number"
        class="form-input optimizedCheckout-form-input"
        data-test="routingNumber-text"
        id="routingNumber"
        maxlength="9"
        name="routingNumber"
        type="tel"
        value=""
      />
    </div>
    <div
      class="form-field"
    >
      <label
        class="form-label optimizedCheckout-form-label"
        for="accountType"
        id="accountType-label"
      >
        optimized_checkout.payment.bluesnap_direct_account_type.label
      </label>
      <div
        class="dropdown-chevron"
      >
        <div
          class="icon"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </div>
      </div>
      <select
        aria-labelledby="accountType-label accountType-field-error-message"
        class="form-select optimizedCheckout-form-select"
        data-test="accountType-select"
        id="accountType"
        name="accountType"
      >
        <option
          selected=""
          value=""
        >
          Select an account type
        </option>
        <option
          value="CONSUMER_CHECKING"
        >
          Consumer checking
        </option>
        <option
          value="CONSUMER_SAVINGS"
        >
          Consumer savings
        </option>
        <option
          value="CORPORATE_CHECKING"
        >
          Corporate checking
        </option>
        <option
          value="CORPORATE_SAVINGS"
        >
          Corporate savings
        </option>
      </select>
    </div>
    <div
      class="form-field"
    >
      <input
        class="form-checkbox optimizedCheckout-form-checkbox"
        id="shopperPermission"
        name="shopperPermission"
        type="checkbox"
      />
      <label
        class="form-label optimizedCheckout-form-label"
        for="shopperPermission"
      >
        optimized_checkout.payment.bluesnap_direct_permission
      </label>
    </div>
  </div>
</fieldset>
`;
